crates/cli/src/plugin.rs

```diff
@@ -0,0 +1,31 @@
+use crate::codegen::plugin::Plugin;
```

Comment:
I don't love this... maybe I would hate it less after we split out the `codegen` crate but at the moment it feels really awkward to have two basically identical structs laying around. Open to suggests here we really only need this wrapper for generating the `JsConfig` :|.

Reply:
Yeah it's a little awkward at the moment but it'll help shrink the size of the next change so I'm in favour.